.github/workflows/build-lint-test.yml

```diff
@@ -118,31 +127,32 @@ jobs:
 
   compatibility-test:
     name: Compatibility test
+    needs: prepare
     runs-on: ubuntu-latest
-    needs:
-      - prepare
     strategy:
       matrix:
         node-version: [18.x, 20.x]
     steps:
-      - name: Use Node.js
+      - uses: actions/checkout@v4
+      - name: Install Corepack via Node
         uses: actions/setup-node@v4
```

Comment:
I didn't find evidence in the [source](https://github.com/actions/setup-node) that it installs dependencies — I think you still have to do that yourself. 

[It _does_ run `yarn` to figure out how to cache Yarn files](https://github.com/actions/setup-node/blob/eff380dfbcf941bf8832e4acb788cebe13dfd758/src/cache-utils.ts#L44), and Yarn v1 is preinstalled on the GitHub runners, and [Yarn v1 throws if `packageManager` refers to a non-1.x release](https://github.com/yarnpkg/yarn/pull/9009). So you're right about that. But... `setup-node` should only try to run `yarn` if you pass the `cache` option to `setup-node`. I left that off the first `setup-node` call as you'd done and I re-run `setup-node` with caching after installing Corepack. So I think that should be okay, but do you still see a case in which this could fail?

As for your final point:

> Could potentially cause issues with any dependencies that rely on the specific package manager version in their installation step

Does Yarn consult `packageManager` for dependencies? I wasn't aware that it did that. I thought it only consulted it for the root package.